components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx

```diff
@@ -79,19 +89,40 @@ export default class PostAttachmentOpenGraph extends React.PureComponent {
             const imageUrl = this.getBestImageUrl(nextProps.openGraphData);
             const {metadata} = nextProps.post;
             const hasLargeImage = metadata && metadata.images && metadata.images[imageUrl] && imageUrl ? this.hasLargeImage(metadata.images[imageUrl]) : false;
+            if (!nextProps.post.metadata && imageUrl) {
```

Comment:
And this to `componentDidUpdate`? If you do that, you can also check if `this.state.imageUrl` has changed when doing that since right now, it'll re-load the image whenever anything changes in the post